### DISCUSSION TITLE
Using git instaweb on a Mac | Loominate

### DISCUSSION THREAD

**Discussion by giscus** [michaelgwelch/loominate.net#10] [Blog Comments]
# Using git instaweb on a Mac | Loominate

The personal blog of Michael Welch. Rarely updated. It contains random exercises and thoughts I had mostly about programming.

https://loominate.net/2012/05/06/using-git-instaweb-on-a-mac/

**Comment by michaelgwelch** [comment]
Just tried running this. Just like years ago I didn't have `lighttpd` installed. Once it was installed I could view my repo. However, what wasn't mentioned above is that you first see a list of projects with only one project that says

> Unnamed repository; edit this file 'description' to name the repository.

Something like this:

<img width="717" alt="image" src="https://user-images.githubusercontent.com/826112/217128961-1f71cb07-ee1f-407c-b340-d84fc36edd28.png">

You can click on that "unnamed repository" and then see your local repo.

If you'd like that to fix the description using your editor in you local working director to edit `.git/description`. Enter a description for your repo (like the name or a short phrase) and then re-run instaweb. You'll now see the name now

<img width="809" alt="image" src="https://user-images.githubusercontent.com/826112/217129273-cecbba45-fed4-4ff6-9530-198470938cec.png">

**Reply by michaelgwelch** [reply]
Some other things I learned. To stop the web server run 

```bash
git instaweb --stop
```

Which is useful if you'd like to look at a different repo.
